### PR TITLE
tooling: add black and flake8 for python formatting/linting

### PR DESCRIPTION
**Pull request by MasterOdin**
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).

This PR adds [black](https://pypi.org/project/black/) and [flake8](https://pypi.org/project/flake8/) to the repo to handle Python code style.

The two recent(ish) PRs #6172 and #6184 included a lot of reformatting of the files unrelated to the change. The purpose was to follow the pep8 standard, plus some stuff from black (like the double quotes). While this is good in the style of better following of Python best practices, mixing it into a PR and not codifying it within CI is just going to lead to style inconsistencies around the Python code. To combat this, this PR adds two popular tools where black handles auto-formatting the code and then flake8 handles linting (there is overlap between the two, with flake8 linting more stuff than black will auto-format).

The big change for people is that they'll need to have Python installed and run `python3 -m pip install -r requirements.txt` before being able to commit. I was not sure what CONTRIBUTING document to put that in. Perhaps a new "Setup" section in https://github.com/tldr-pages/tldr/blob/main/CONTRIBUTING.md like follows:

> ## Setup
>
> The tldr repo uses a handful of automated tools to check styling of the markdown files and
> build scripts. To use these tools, you will need installed on your computer:
>
> * Python 3.6+
> * NodeJS 12+
> 
> After installing the above, you can get the automated tools by running:
>
> * `npm install`
> * `python3 -m pip install -r requirements.txt`
>
> All tools are run automatically when you run `git commit`.

### CODE REVIEW

File: scripts/pdf/render.py [hunk 1]
```diff
@@ -15,7 +15,8 @@
 import argparse
 from datetime import datetime
 
-from weasyprint import HTML, CSS
+from weasyprint import HTML
+
```
Comment (marchersimon) [inline scripts/pdf/render.py:19]: Why the extra empty line?

Comment (MasterOdin) [reply scripts/pdf/render.py:19]: Per pep8, there should be two empty lines before any function (and class) definition.

Functions within a class only need one empty like before the function definition (except the first function).

**Review by marchersimon — APPROVED** [review]
For the most part I don't know what's going on here, but have my approval, since it's probably alright. Thanks!

**Comment by blueskyson** [comment]
Thanks for adding the formatting stuff! Will `black` and `flake8` format python scripts automatically when executing `test.py`, or do we need to run something like `black file.py & flake8 file.py`?

**Comment by navarroaxel** [comment]
@MasterOdin could you please resolve the conflicts?

**Comment by MasterOdin** [comment]
@navarroaxel done

**Review by owenvoke — APPROVED** [review]

**Review by navarroaxel — APPROVED** [review]

**Pull request merged**